### ISSUE TITLE
[Bug] Reachable assertion in get_layer4_v6

### ISSUE THREAD

**Issue by kdsjZh** [appneta/tcpreplay#717]
You are opening a _bug report_ against the Tcpreplay project: we use
GitHub Issues for tracking bug reports and feature requests.

If you have a question about how to use Tcpreplay, you are at the wrong
site. You can ask a question on the [tcpreplay-users mailing list](https://sourceforge.net/p/tcpreplay/mailman/tcpreplay-users/)
or [on Stack Overflow with  [tcpreplay] tag](http://stackoverflow.com/questions/tagged/tcpreplay). 
General help is available [here](http://tcpreplay.appneta.com/).

If you have a build issue, consider downloading the [latest release](https://github.com/appneta/tcpreplay/releases/latest)

Otherwise, to report a bug, please fill out the reproduction steps
(below) and delete these introductory paragraphs.  Thanks!

**Describe the bug**
The assertion `assert(ip6_hdr);` in `get_layer4_v6()` at `common/get.c` is reachable when the user uses tcprewrite to open a crafted pcap file.

**To Reproduce**
1. export CC=clang && export CFLAGS="-fsanitize=address -g"
2. ./autogen.sh && ./configure --disable-shared --disable-local-libopts && make clean && make -j8
3. src/tcprewrite -o /dev/null -i POC
output:
```
Warning: tcprewrite/crash.0 was captured using a snaplen of 96 bytes.  This may mean you have truncated packets.
tcprewrite: get.c:599: void *get_layer4_v6(const ipv6_hdr_t *, const int): Assertion `ip6_hdr' failed.
Aborted
```
**Expected behavior**
Program reports assertion failure and is terminated.

**Screenshots**
![image](https://user-images.githubusercontent.com/35988108/156726518-b1764ba1-3368-46e4-a44a-c9e04a0d4221.png)

**System (please complete the following information):**
 - OS: Ubuntu
 - OS version : can be reproduced in 18.04/20.04
 - Clang version : clang-12.0.1 (release/12.x)
 - Tcpreplay Version : latest commit 09f07748dcabe3d58961f123f31dd0f75198a389

**Credit**
[NCNIPC of China](http://www.nipc.org.cn)
[Hexhive](http://hexhive.epfl.ch/)

**Comment by kdsjZh** [comment]
[POC1.zip](https://github.com/appneta/tcpreplay/files/8184223/POC1.zip)

**Comment by fklassen** [comment]
Fixed in PR #739. Added test for null pointer.
